server/datastore/mysql/software.go

```diff
@@ -196,11 +196,9 @@ func getOrGenerateSoftwareIdDB(ctx context.Context, tx sqlx.ExtContext, s fleet.
 	if err != nil {
 		return 0, ctxerr.Wrap(ctx, err, "insert software")
 	}
-	id, err := result.LastInsertId()
-	if err != nil {
-		return 0, ctxerr.Wrap(ctx, err, "last id from software")
-	}
-	return uint(id), nil
+	// LastInsertedId sometimes returns 0 as it's dependent on connections and how mysql is configured
```

Comment:
Should we never trust `LastInsertId` then?

typos:
- `// LastInsertId ...`
- It seems the comment is missing a period at the end of the line? (I was confused at first.)

Reply:
Or does it return 0 when it updates instead of inserting (`... ON DUPLICATE KEY UPDATE`)

Reply:
Also, this happened with Linux hosts, so bundle identifier should be the same, and so it shouldn't end up here I don't think.